optimade/server/models.py

```diff
@@ -53,24 +58,110 @@ class StructureResource(EntryResource):
+                    "pointing to homepage of the database provider, either "
+                    "directly as a string, or as a link object."
+    )
+
+    index_base_url: Optional[Union[UrlStr, Link]] = Schema(
+        ...,
+        description="a [JSON API links object](http://jsonapi.org/format/#document-links) "
+                    "pointing to the base URL for the `index` meta-database as "
+                    "specified in Appendix 1, either directly as a string, or "
+                    "as a link object."
+    )
+
+
+class OptimadeResponseMeta(BaseModel):
     """
-    query: OptimadeResponseMetaQuery
-    api_version: str = Schema(..., description="a string containing the version of the API implementation.")
-    time_stamp: datetime
-    data_returned: NonnegativeInt
-    more_data_available: bool
+    A [JSON API meta member](https://jsonapi.org/format/#document-meta)
+    that contains JSON API meta objects of non-standard
+    meta-information.
+
+    OPTIONAL additional information global to the query that is not
+    specified in this document, MUST start with a
+    database-provider-specific prefix.
+
+    """
+
+    query: OptimadeResponseMetaQuery = Schema(
+        ...,
+        description="information on the query that was requested"
+    )
+
+    api_version: str = Schema(
+        ...,
+        description="a string containing the version of the API "
+                    "implementation"
+    )
+
+    time_stamp: datetime = Schema(
+        ...,
+        description="a string containing the date and time at which "
+                    "the query was exexcuted, in "
+                    "[ISO 8601](https://www.iso.org/standard/40874.html) "
+                    "format. Times MUST be time-zone aware (i.e. MUST "
+                    "NOT be local times), in one of the formats allowed "
+                    "by ISO 8601 (i.e. either be in UTC, and then end "
+                    "with a Z, or indicate explicitly the offset)."
+    )
+
+    data_returned: NonnegativeInt = Schema(
+        ...,
+        description="an integer containing the number of data objects "
+                    "returned for the query."
+    )
+
+    more_data_available: bool = Schema(
+        ...,
+        description="`false` if all data has been returned, and `true` "
+                    "if not."
+    )
+
+    provider: OptimadeProvider = Schema(
+        ...,
+        description="information on the database provider of the implementation."
+    )
+
+    data_available: Optional[int] = Schema(
+        ...,
+        description="an integer containing the total number of data "
+                    "objects available in the database"
+    )
+
+    last_id: Optional[str] = Schema(
+        ...,
+        description="a string containing the last ID returned"
+    )
+
+    response_message: Optional[str] = Schema(
+        ...,
+        description="response string from the server"
```

Comment:
do we say what this is?
I guess it won't be an error and it won't be warnings (for which there are specific information channels).
What is a "response string"?

Reply:
Doesn't say anything more in the spec!